backend/src/main/java/com/map/repository/UserRepository.java

```diff
@@ -63,4 +65,42 @@ public List<User> getAllUsers() {
         }
         return users;
     }
-}
+
+    public User getUserByNickname(String userNickname) {
+            User user = new User();
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement("select * from map.user where nickname=?");
+            preparedStatement.setString(1, userNickname);
+            ResultSet rs = preparedStatement.executeQuery();
+            if (rs.next()) {
+                user.setNickname(rs.getString("nickname"));
+                user.setCity(rs.getString("city"));
+                user.setCountry(rs.getString("country"));
+                user.setZipCode(rs.getString("zipCode"));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return user;
+    }
+
+    private void checkIfUserExists(String userNickname) {
```

Comment:
I think it could be returning boolean, because you expect to find or not user, I don't see need for throwing exception here.

Reply:
If there is no user, ResulSet will be empty.